### PR TITLE
Fix/API Version

### PR DESCRIPTION
**Pull request by samchristensen**

### CODE REVIEW

**Review by caioariede — COMMENTED** [review]
What do you think about this approach, vs being able to pass custom settings?

For example, instead of accessing `settings`, it could access `self.settings`, and then we could set it as:

```python
adyen = Adyen()
adyen.client.settings = custom_settings
```

**Comment by samchristensen** [comment]
I tried that approach, but I thought it introduced more changes to the code that we wanted. Specifically because it has more side effects than just an optional argument.

the way they currently have their client structured, it defaults to `settings.API_PAYMENT_VERSION`, but in reality if someone was going to make several calls to the API and want to use different versions, they would have to reinstantiate the client every time. 

obviously we don't use a client object more than once currently, but i'm trying to keep it general since others out there might

**Comment by samchristensen** [comment]
@caioariede I think that approach makes sense if the `def _determine_api_url(...)` function wasn't structured how it is, but since it's structured to fallback to a specific setting value, i think that approach isn't as flexible as we want

**Comment by caioariede** [comment]
Okay, I still think we should try to see the problem in a higher level. I don't think it makes sense to mix versions for the same service.

One approach I can see, is leveraging settings load so it can be changed to something else, like:

```python
from . import settings
```

to

```python
settings = __import__(os.environ.get("ADYEN_SETTINGS_MODULE", "Adyen.settings"))
```

Thoughts?

**Comment by samchristensen** [comment]
@caioariede I agree that it doesn't make sense to mix versions, but that's how they are doing it currently. That's what i'm saying, is that I agree that the ideal setup would be to not mix versions and to use something like settings, but they are already mixing versions in their SDK.

For instance, this function will return potentially 3 or 4 different versions, depending on which API you are calling 

https://github.com/Adyen/adyen-python-api-library/blob/278ef674c2ac801a4546c8fbddf9b421f649f032/Adyen/client.py#L96

**Comment by caioariede** [comment]
@samchristensen when I say about mixing versions, I say about mixing versions for the same service. Having different versions for each service is understandable

**Comment by samchristensen** [comment]
@caioariede how about instead of doing `import` of the settings, we do something like passing them into the client object? I don't like the idea of having to set and environment variable to override the settings

**Comment by samchristensen** [comment]
@caioariede I will make the change and see what you think

**Comment by samchristensen** [comment]
@caioariede the intention with this setup is to allow for us to do the following to override the settings

```
class CustomAdyenClientSettings(AdyenClientSettings):
    API_BIN_LOOKUP_VERSION = 'v50'

....

self.client = AdyenClient(settings=CustomAdyenClientSettings)
```

File: Adyen/settings.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-ENDPOINT_CHECKOUT_LIVE_SUFFIX = "https://{}-checkout-live" \
-                                ".adyenpayments.com/checkout"
-API_CHECKOUT_VERSION = "v49"
-API_CHECKOUT_UTILITY_VERSION = "v1"
-API_RECURRING_VERSION = "v25"
-API_PAYMENT_VERSION = "v49"
-API_PAYOUT_VERSION = "v30"
-LIB_VERSION = "2.1.0"
-LIB_NAME = "adyen-python-api-library"
+class AdyenClientSettings(object):
```
Comment (caioariede) [inline Adyen/settings.py]: Are this settings specific to `AdyenClient` or they are used elsewhere?

Comment (samchristensen) [reply Adyen/settings.py]: at the moment, they only use them here, but i don't want to remove it because i don't want to change too much about their code

Comment (caioariede) [reply Adyen/settings.py]: I think we shouldn't do any changes in this file

File: Adyen/client.py [hunk 2]
```diff
@@ -72,7 +73,7 @@ def __init__(self, username=None, password=None, xapikey=None,
                  platform="test", merchant_account=None,
                  merchant_specific_url=None, skin_code=None,
                  hmac=None, app_name="",
-                 http_force=None, live_endpoint_prefix=None):
+                 http_force=None, live_endpoint_prefix=None, settings=None):
```
Comment (caioariede) [inline Adyen/client.py]: this should be `settings=settings` (keeping the original import)

**Review by caioariede — CHANGES_REQUESTED** [review]

File: Adyen/client.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 import json as json_lib
 import re
 
-from . import util
+from . import util, settings
```
Comment (caioariede) [inline Adyen/client.py]: should be `settings, util`

**Review by caioariede — CHANGES_REQUESTED** [review]
minor thing

**Comment by caioariede** [comment]
@samchristensen good work! can we change base of this PR to `Adyen/adyen-python-api-library:develop`?

**Comment by caioariede** [comment]
or you can just close this PR and open a new one pointing to Adyen's repo

**Pull request closed without merging**